packages/sdk/lib/base.ts

```diff
@@ -433,17 +482,7 @@ export class Itly {
         || validationResponses.every((vr) => vr.valid);
     }
 
-    // #2 track phase
-    // invoke track(), group(), identify(), page() on every plugin if allowed
-    if (shouldRun) {
```

Comment:
We still want to run `method()`s before `postMethod()`s but we don't want to wait for them to complete. Instead of moving this below I am think we want something more like:
```
// track phase
const promise = shouldRun
  ? this.runOnAllPlugins(op, (p) => { ... method(p, event)}
  : Promise.resolve();
// postTrack
// validation error
return promise;
```

Reply:
Fixed